opentelemetry/src/global/internal_logging.rs

```diff
@@ -23,12 +23,22 @@ macro_rules! otel_info {
         {
             tracing::info!( name: $name, target: env!("CARGO_PKG_NAME"), "");
         }
+        #[allow(unused_variables)]
+        #[cfg(not(feature = "internal-logs"))]
+        const _: () = {
+            let _ = $name;
+        };
```

Comment:
`const` here will (double) ensure that the rust compiler will optimize out this at compile-time, and so no runtime code generated.